app/frontend/src/features/communities/events/EventPage.tsx

```diff
@@ -0,0 +1,211 @@
+  const isValidEventId = !isNaN(eventId) && eventId > 0;
+  const queryClient = useQueryClient();
+  const {
+    data: event,
+    error: eventError,
+    isLoading,
+  } = useQuery<Event.AsObject, GrpcError>({
+    queryKey: eventKey(eventId),
+    queryFn: () => service.events.getEvent(eventId),
+    enabled: isValidEventId,
+  });
+
+  const {
+    isLoading: isSetEventAttendanceLoading,
+    error: setEventAttendanceError,
+    mutate: setEventAttendance,
+  } = useMutation<Event.AsObject, GrpcError, AttendanceState>(
+    (currentAttendanceState) => {
+      const attendanceStateToSet =
+        currentAttendanceState === AttendanceState.ATTENDANCE_STATE_GOING
+          ? AttendanceState.ATTENDANCE_STATE_NOT_GOING
+          : AttendanceState.ATTENDANCE_STATE_GOING;
+      return service.events.setEventAttendance({
+        attendanceState: attendanceStateToSet,
+        eventId,
+      });
+    },
+    {
+      onSuccess(updatedEvent) {
+        queryClient.setQueryData<Event.AsObject>(
+          eventKey(eventId),
+          updatedEvent
+        );
+        queryClient.invalidateQueries(eventKey(eventId));
+        queryClient.invalidateQueries({
+          predicate: ({ queryKey }) =>
+            queryKey[0] === eventAttendeesBaseKey &&
+            isEventUsersInputType(queryKey[1]) &&
+            queryKey[1].eventId === eventId,
+        });
+      },
+    }
+  );
+
+  useEffect(() => {
+    if (event?.slug && event.slug !== eventSlug) {
+      history.replace(routeToEvent(event.eventId, event.slug));
+    }
+  }, [event, eventSlug, history]);
+
+  return !isValidEventId ? (
+    <NotFoundPage />
+  ) : (
+    <>
+      {(eventError || setEventAttendanceError) && (
+        <Alert severity="error">
+          {eventError?.message || setEventAttendanceError?.message || ""}
+        </Alert>
+      )}
+      {isLoading ? (
+        <CircularProgress />
+      ) : (
+        event && (
+          <>
+            <div className={classes.header}>
+              <HeaderButton
+                className={classes.backButton}
+                onClick={() => history.goBack()}
+              >
+                <BackIcon />
+              </HeaderButton>
+              <div className={classes.eventTitle}>
+                <Typography variant="h1">{event.title}</Typography>
+                <Typography className={classes.eventTypeText} variant="body1">
+                  {event.onlineInformation
+                    ? VIRTUAL_EVENT
+                    : event.offlineInformation?.address}
+                </Typography>
+              </div>
+              <Button
+                className={classes.attendanceButton}
+                loading={isSetEventAttendanceLoading}
+                onClick={() => setEventAttendance(event.attendanceState)}
+              >
+                {event.attendanceState ===
+                AttendanceState.ATTENDANCE_STATE_GOING
+                  ? NOT_GOING
+                  : GOING}
+              </Button>
```

Comment:
To me this isn't clear if I'm attending or not and what clicking the button does.
I think you should make the labels "Join event" and "Leave event", and make the latter an outlined variant.
Could even optionally have a line somewhere "You are attending this event." if so

Reply:
I'll change the label, those were copied from the design. I think the "you are attending" subtext may not be needed as it will hopefully be more obvious when the API bug is fixed so your user is actually shown correctly under attendees!